sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/QueryResult.cs

```diff
@@ -54,6 +52,21 @@ internal static QueryResult<T> DeserializeQueryResult(JsonElement element, Objec
         {
             IReadOnlyList<T> items = default;
             string continuationToken = default;
+
+            // If the provided objectSerializer is of type JsonObjectSerializer, we will re-use the same object and set it as the defaultObjectSerializer.
+            if (objectSerializer is JsonObjectSerializer)
```

Comment:
If we know that objectSerializer will always be a JsonObjectSerliazer, no matter what is passed to the DeserializeQueryResult, can't we get rid of the objectSerializer parameter for the method and use a static defaultObjectSerializer that is initialized at constructor?

Reply:
We still have to use that provided serializer to deserialize the stream into the user specified type. which may be a different serializer than the JsonObjectSerializer